src/components/YourModule.js

```diff
@@ -52,58 +51,28 @@ export default class YourModule extends React.Component {
   }
 
   bet(outcome) {
-    //Gnosis.create(
-    //  {ethereum: this.props.xdaiweb3.provider}
-    //).then(async (result) => {
-      console.log("BETTING...")
-      //const market =  result.contracts.Market.at(this.state.address);
       const cost = Gnosis.calcLMSROutcomeTokenCount(
           this.state.outcomeTokensSold,
           1000e18,
           outcome,
           this.state.amount * 1e18,
           0
       );
-      console.log("CALL buy(",outcome,",",cost.toNumber(),",",10 * 1e18,")")
-      //console.log("market:",market)
-      //console.log("Getting txn...")
-      //let txn = await market.buy(outcome, cost.toNumber(), 10 * 1e18)
-      //console.log("TXN",txn)
-      //let txnresult = await this.props.tx(txn, 50000, 0, 0)
-      //console.log("txnresult:",txnresult)
-      /*
-      "inputs": [
-  			{
-  				"name": "outcomeTokenIndex",
-  				"type": "uint8"
-  			},
-  			{
-  				"name": "outcomeTokenCount",
-  				"type": "uint256"
-  			},
-  			{
-  				"name": "maxCost",
-  				"type": "uint256"
-  			}
-  		],
-       */
       this.props.tx(
-        this.props.contracts.Market.buy(outcome,cost.toNumber(),10 * 1e18)
-        ,240000,0,0,(receipt)=>{
-          if(receipt){
-
-            console.log("BET COMPLETE?!?",receipt)
-            //this.props.goBack();
-          //  window.history.pushState({},"", "/");
-            //this.props.setReceipt({to:toAddress,from:receipt.from,badge:this.props.badge,result:receipt})
-            //this.props.changeView("receipt");
-            //this.props.clearBadges()
-          }
-        }
-      )
-    //})
-
+        this.props.contracts.ERC20Vendable.approve(this.props.contracts.Market._address, -1),
```

Comment:
This transaction is being done always, while should be done only once per commit.

@austintgriffith How can we make calls to allowance to check if we need to make it? 

Would be enough just making `this.props.ERC20Vendable.allowance`?